acme/acme/client.py

```diff
@@ -259,12 +262,16 @@ def retry_after(cls, response, default):
         try:
             seconds = int(retry_after)
         except ValueError:
-            # pylint: disable=no-member
-            decoded = werkzeug.parse_date(retry_after)  # RFC1123
-            if decoded is None:
-                seconds = default
-            else:
-                return decoded
+            # The RFC 2822 parser handles all of RFC 2616's cases in modern
+            # environments (primarily HTTP 1.1+ but also py27+)
+            when = parsedate_tz(retry_after)
```

Comment:
`werkzeug` and you in a previous version of this PR call `strip` on `retry_after` ([source](https://github.com/letsencrypt/letsencrypt/blob/6fd3dba737a8de2c2f2d43d52d932418762762bb/acme/acme/client.py#L259)). I think we should be doing this unless we are **certain** it isn't necessary.